### PR TITLE
Fix #10455 Allow use of node version 20

### LINKED ISSUE #10455: Allow use of node version 20
## Description

After some test it seem possible to use node 20 by adding the following flag to .npmrc
```
node-options=--openssl-legacy-provider
```

**What kind of improvement you want to add?** (check one with "x", remove the others)
- [x] Minor changes to existing features

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

~~This PR adds the `node-options=--openssl-legacy-provider` variable to the `.npmrc` files to use node version 20.~~

This PR updates webapck, html-webpack-plugin and html-loader dependecies and configures a different hashFunction for webapck to allow usage of node v20

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Minor changes to existing features
 

## Issue

**What is the current behavior?**

#10455

**What is the new behavior?**

It is possible to install, run and compile mapstore with node version 20

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] Yes

### CODE REVIEW

**Comment by allyoucanmap** [comment]
@tdipisa @offtherailz unfortunately it's not working without updating also the github actions.
We should also document someway this update but I'm not sure the migration guidelines is the right place

**Review by offtherailz — COMMENTED** [review]
I updated the documentation with this [commit](https://github.com/geosolutions-it/MapStore2/pull/10457/commits/3dafbc988cbb8dfa37e8ed8f728570d1f608fbe9) on the PR
Note @tdipisa and @allyoucanmap This is a **breaking change**. When this PR is merged,  the application will not be able to build anymore with current version of node used.
I didn't found any workaround except to update the dependencies.
We can merge this PR and **update all our CI/CD** chain for this breaking change. Also projects, when updated, will need this upgrade of node version.

As an alternative we should identfy what libraries need this update, and evaluate the cost of the upate. 

Please @allyoucanmap and @tdipisa let's check this and, before merging this PR, be aware of the changes needed.

**Comment by allyoucanmap** [comment]
> I updated the documentation with this [commit](https://github.com/geosolutions-it/MapStore2/pull/10457/commits/3dafbc988cbb8dfa37e8ed8f728570d1f608fbe9) on the PR Note @tdipisa and @allyoucanmap This is a **breaking change**. When this PR is merged, the application will not be able to build anymore with current version of node used. I didn't found any workaround except to update the dependencies. We can merge this PR and **update all our CI/CD** (e.g. jenkins) chain for this breaking change. Also projects, when updated, will need this upgrade of node version.
> 
> As an alternative we should identfy what libraries need this update, and evaluate the cost of the upate.
> 
> Please @allyoucanmap and @tdipisa let's check this and, before merging this PR, be aware of the changes needed.

@offtherailz @tdipisa I double checked and it was the hash function used by webapack that was causing the problem, after updating webapck, html-webpack-plugin and html-loader dependecies it's possible to configure a different hashFunction that support newer and current version of node. I tested node v16.20.1 and v20.13.1 both are working without the --openssl-legacy-provider flag.

File: docs/developer-guide/mapstore-migration-guide.md [hunk 1]
```diff
@@ -22,6 +22,13 @@ This is a list of things to check if you want to update from a previous version
 
 ## Migration from 2024.01.02 to 2024.02.00
 
+### NodeJS and NPM update
+
+From this version the **recommended version** to build MapStore or for development is **Node 20** (**minimum** version will be **Node 17**).
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:27]: Minimum version is 16 according with requirements. 

```suggestion
From this version the **recommended version** to build MapStore or for development is **Node 20** (**minimum** version will be **Node 16**).
```

File: docs/developer-guide/requirements.md [hunk 2]
```diff
@@ -17,11 +17,11 @@ These tools needs to be installed (other than **Java** in versions above above):
 
 | Tool                  | Link                                                       | Minimum | Recommended | Maximum             |
 |-----------------------|------------------------------------------------------------|---------|-------------|---------------------|
-| npm                   | [link](https://www.npmjs.com/get-npm)                      | 8       | 8           | 8.19.4<sup>3</sup>  |
-| NodeJS                | [link](https://nodejs.org/en/)                             | 16      | 16          | 16.20.1<sup>3</sup> |
+| npm                   | [link](https://www.npmjs.com/get-npm)                      | 8.11.0  | 10.1.0      |                     |
+| NodeJS                | [link](https://nodejs.org/en/)                             | 17.9.1  | 20          |                     |
```
Comment (allyoucanmap) [inline docs/developer-guide/requirements.md:21]:
```suggestion
| NodeJS                | [link](https://nodejs.org/en/)                             | 16.20.1  | 20          |                     |
```

**Review by offtherailz — APPROVED** [review]
LGTM.
We have jenkins using node 16 so we can test also the minimum version twice. 

We have then to update it at suggested version, if we want better performances etc...

**Comment by allyoucanmap** [comment]
@ElenaGallo @tdipisa there is no specific test for this improvement, we need to verify if DEV will build correctly.

**Pull request merged**